tests/otel_integrations/test_openai.py

```diff
@@ -632,8 +939,8 @@ async def test_async_chat_completions_stream(
                     'logfire.span_type': 'log',
                     'logfire.tags': ('LLM',),
                     'duration': 1.0,
-                    'response_data': '{"combined_chunk_content":"The answer is secret","chunk_count":2}',
-                    'logfire.json_schema': '{"type":"object","properties":{"request_data":{"type":"object"},"async":{},"duration":{},"response_data":{"type":"object"}}}',
+                    'response_data': '{"message":{"content":"The answer is secret","refusal":null,"role":"assistantassistantassistant","audio":null,"function_call":null,"tool_calls":null,"parsed":null},"usage":null}',
```

Comment:
why is the role `assistantassistantassistant`?

Reply:
In the hardcoded test response the "role" field should be null except for in the first chunk. Fixed now in https://github.com/pydantic/logfire/pull/545/commits/7045b77b916731dde33dc9ed60658f8fbc27f65a

```python
from openai import Client

response = Client().chat.completions.create(
    model="gpt-4",
    messages=[{"role": "user", "content": "Just say: The answer is secret."}],
    stream=True,
    # stream_options={"include_usage": True},
    # n=2,
)
for chunk in response:
    print(chunk.model_dump_json(exclude_none=True))

{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{"content":"","role":"assistant"},"index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{"content":"The"},"index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{"content":" answer"},"index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{"content":" is"},"index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{"content":" secret"},"index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{"content":"."},"index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
{"id":"chatcmpl-ASBkUgeHXBMarEwWnFLbT4vYMsldh","choices":[{"delta":{},"finish_reason":"stop","index":0}],"created":1731281710,"model":"gpt-4-0613","object":"chat.completion.chunk"}
```

Reply:
Still seems like the roles shouldn't be concatenated like that when they're present in the response data. Is that concatenation in our code, or openai's?

Reply:
This behavior is from the openai code. They update the current snapshot of the ChatCompletion message by treating it as a dict and determining how to update each field based on its type. There is currently no special handling of the "role" field -- strings are concatenated.
https://github.com/openai/openai-python/blob/646a579cdb305a9d3fba6c5f9a96011c5e2c2882/src/openai/lib/streaming/_deltas.py#L27-L28

Reply:
ollama "openai-compatible" API provides the role in every message so is not compatible with this. I'll open an issue with them.

```python
from openai import Client

client = Client(
    base_url="http://localhost:11434/v1",
    api_key="ollama",
)
response = client.chat.completions.create(
    model="llama3.1",
    messages=[{"role": "user", "content": "Just say: The answer is secret."}],
    stream=True,
)
for chunk in response:
    print(chunk.model_dump_json(exclude_none=True))
{"id":"chatcmpl-804","choices":[{"delta":{"content":"The","role":"assistant"},"index":0}],"created":1731398387,"model":"llama3.1","object":"chat.completion.chunk","system_fingerprint":"fp_ollama"}
{"id":"chatcmpl-804","choices":[{"delta":{"content":" answer","role":"assistant"},"index":0}],"created":1731398388,"model":"llama3.1","object":"chat.completion.chunk","system_fingerprint":"fp_ollama"}
{"id":"chatcmpl-804","choices":[{"delta":{"content":" is","role":"assistant"},"index":0}],"created":1731398388,"model":"llama3.1","object":"chat.completion.chunk","system_fingerprint":"fp_ollama"}
{"id":"chatcmpl-804","choices":[{"delta":{"content":" secret","role":"assistant"},"index":0}],"created":1731398388,"model":"llama3.1","object":"chat.completion.chunk","system_fingerprint":"fp_ollama"}
{"id":"chatcmpl-804","choices":[{"delta":{"content":".","role":"assistant"},"index":0}],"created":1731398388,"model":"llama3.1","object":"chat.completion.chunk","system_fingerprint":"fp_ollama"}
{"id":"chatcmpl-804","choices":[{"delta":{"content":"","role":"assistant"},"finish_reason":"stop","index":0}],"created":1731398388,"model":"llama3.1","object":"chat.completion.chunk","system_fingerprint":"fp_ollama"}
```

The message still appears in the UI with the concatenated role.

<img width="404" alt="image" src="https://github.com/user-attachments/assets/daf808da-dc61-461d-8e4e-684f6611663f">